sunshine/src/main/java/digital/and/sunshine/location/LocationSeekerRandomAdapter.java

```diff
@@ -46,8 +48,8 @@ private Coordinates generateCoordinates(final Coordinates coordinates, final int
     final var dy = r * Math.sin(theta);
     final var dx = r * Math.cos(theta);
 
-    final var newLatitude = coordinates.lat() + dy / DEGREE;
-    final var newLongitude = coordinates.lon() + dx / (DEGREE * Math.cos(toRadians(coordinates.lat())));
+    final var newLatitude = new BigDecimal(coordinates.lat() + dy / DEGREE).setScale(6, RoundingMode.HALF_UP).doubleValue();
+    final var newLongitude = new BigDecimal(coordinates.lon() + dx / (DEGREE * Math.cos(toRadians(coordinates.lat())))).setScale(6, RoundingMode.HALF_UP).doubleValue();
```

Comment:
You fixed double Precision Problems. Perfect. 
Maybe We can use BigDecimal.valueOf instead of constructor. @brunasti

Reply:
Good point, let me try :-)